pkg/kubelet/cm/topologymanager/numatopology/numa_topology_test.go

```diff
@@ -0,0 +1,348 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package numatopology
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	cadvisorapi "github.com/google/cadvisor/info/v1"
+
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager/bitmask"
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager/numatopology/fakesysfs"
+)
+
+func TestNUMATopology(t *testing.T) {
+	tcases := []struct {
+		name             string
+		topology         []cadvisorapi.Node
+		expectedTopology Topology
+		expectedErr      error
+	}{
+		{
+			name: "positive test 1 node",
+			topology: []cadvisorapi.Node{
+				{
+					Id: 0,
+				},
+			},
+			expectedTopology: Topology{
+				numaNodes: []int{0},
+				distances: [][]uint64{
+					{
+						10,
+						11,
+						12,
+						12,
+					},
+				},
+			},
+		},
+		{
+			name: "positive test 2 nodes",
+			topology: []cadvisorapi.Node{
+				{
+					Id: 0,
+				},
+				{
+					Id: 1,
+				},
+			},
+			expectedTopology: Topology{
+				numaNodes: []int{0, 1},
+				distances: [][]uint64{
+					{
+						10,
+						11,
+						12,
+						12,
+					},
+					{
+						11,
+						10,
+						12,
+						12,
+					},
+				},
+			},
+		},
+		{
+			name: "positive test 3 nodes",
+			topology: []cadvisorapi.Node{
+				{
+					Id: 0,
+				},
+				{
+					Id: 1,
+				},
+				{
+					Id: 2,
+				},
+			},
+			expectedTopology: Topology{
+				numaNodes: []int{0, 1, 2},
+				distances: [][]uint64{
+					{
+						10,
+						11,
+						12,
+						12,
+					},
+					{
+						11,
+						10,
+						12,
+						12,
+					},
+					{
+						12,
+						12,
+						10,
+						11,
+					},
+				},
+			},
+		},
+		{
+			name: "positive test 4 nodes",
+			topology: []cadvisorapi.Node{
+				{
+					Id: 0,
+				},
+				{
+					Id: 1,
+				},
+				{
+					Id: 2,
+				},
+				{
+					Id: 3,
+				},
+			},
+			expectedTopology: Topology{
+				numaNodes: []int{0, 1, 2, 3},
+				distances: [][]uint64{
+					{
+						10,
+						11,
+						12,
+						12,
+					},
+					{
+						11,
+						10,
+						12,
+						12,
+					},
+					{
+						12,
+						12,
+						10,
+						11,
+					},
+					{
+						12,
+						12,
+						11,
+						10,
+					},
+				},
+			},
+		},
+		{
+			name: "negative test 1 node",
+			topology: []cadvisorapi.Node{
+				{
+					Id: 0,
+				},
+			},
+			expectedTopology: Topology{},
+			expectedErr:      fmt.Errorf("no distance file found"),
+		},
+	}
+
+	fakeSysFs := fakesysfs.FakeSysFs{}
+	fakeSysFs.SetDistances(0, "10 11 12 12", nil)
+	fakeSysFs.SetDistances(1, "11 10 12 12", nil)
+	fakeSysFs.SetDistances(2, "12 12 10 11", nil)
+	fakeSysFs.SetDistances(3, "12 12 11 10", nil)
```

Comment:
I'd prefer not to use an abstracton that we plan on eventually removing inside these tests. Once we move to the cadvisor api, we should be able to simply remove the `numatopology/sysfs` abstraction you have and not modify anything about the rest of these files.